auslib/web/admin/views/rules.py

```diff
@@ -89,40 +103,37 @@ def _post(self, id_or_alias, transaction, changed_by):
         rule = dbo.rules.getRule(id_or_alias, transaction=transaction)
         if not rule:
             return Response(status=404)
-        form = EditRuleForm()
 
-        releaseNames = dbo.releases.getReleaseNames()
+        edit_rule_dict, mapping_values = process_rule_form(connexion.request.json)
 
-        form.mapping.choices = [(item['name'], item['name']) for item in releaseNames]
-        form.mapping.choices.insert(0, ('', 'NULL'))
+        if edit_rule_dict.get('mapping', None) is not None and len(mapping_values) != 1:
+            return problem(400, 'Bad Request', 'Invalid mapping value. No release name found in DB')
 
-        if not form.validate():
-            self.log.warning("Bad input: %s", form.errors)
-            return Response(status=400, response=json.dumps(form.errors))
+        # Solves https://bugzilla.mozilla.org/show_bug.cgi?id=1361158
+        edit_rule_dict.pop("csrf_token", None)
 
         what = dict()
         # We need to be able to support changing AND removing parts of a rule,
         # and because of how Flask's request object and WTForm's defaults work
         # this gets a little hary.
-        for k, v in form.data.iteritems():
+        for k, v in edit_rule_dict.iteritems():
```

Comment:
I think you can get rid of this entire block if you pop data_version like you're doing above for csrf_token. Now that the request.json/request.form handling is gone, the only thing this block does is filter out data_version.